### PR TITLE
Distinguish between RealizationState and RealizationStorageState in SimulationContext

### PR DESCRIPTION
**Pull request by xjules**
Distinguish between RealizationState and RealizationStorageState in SimulationContext

### CODE REVIEW

**Review by berland — APPROVED** [review]
👍🏻

File: src/ert/simulator/simulation_context.py [hunk 1]
```diff
@@ -68,12 +68,15 @@ def _run_forward_model(
...
-                job_queue.stop_long_running_jobs,
+                job_queue.stop_long_running_realizations,
                 ert.ert_config.analysis_config.minimum_required_realizations,
             )
         ]
 
-    asyncio.run(job_queue.execute(evaluators=queue_evaluators))
+    try:
+        asyncio.run(job_queue.execute(evaluators=queue_evaluators))  # type: ignore
+    except Exception:
```
Comment (berland) [inline src/ert/simulator/simulation_context.py:78]: bare exceptions are not a good idea.  Being bare, it should at least be re-raised?

File: src/ert/simulator/simulation_context.py [hunk 2]
```diff
@@ -68,12 +68,15 @@ def _run_forward_model(
...
+        job_queue.kill_all_jobs()
```
Comment (berland) [inline src/ert/simulator/simulation_context.py:79]: This trickles down to an  asyncio.create_task() in realization_state.py. Will that work outside the async context?

File: src/ert/simulator/simulation_context.py (hunk 1 above)
Comment (berland) [reply src/ert/simulator/simulation_context.py:78]: There is a bare exception catch in queue.py already. Should not be needed.

Comment (xjules) [reply src/ert/simulator/simulation_context.py:78]: Yes, but the point is that this gets raised in here only.

File: src/ert/simulator/simulation_context.py (hunk 2 above)
Comment (xjules) [reply src/ert/simulator/simulation_context.py:79]: That's what I don't know :man_shrugging:

Comment (berland) [reply src/ert/simulator/simulation_context.py:79]: It won't work. You'll get "RunTimeError: no running event loop"

File: src/ert/simulator/simulation_context.py (hunk 1 above)
Comment (berland) [reply src/ert/simulator/simulation_context.py:78]: I think the queue should be responsible for catching the exception(s) here yes.

File: src/ert/simulator/simulation_context.py (hunk 2 above)
Comment (xjules) [reply src/ert/simulator/simulation_context.py]: removed!

**Pull request merged**